File: backend-node/src/models/StreamInfo.ts
import { Track } from './Track';

export enum AudioFormat {
  MP3 = 'MP3',
  AAC = 'AAC',
  WEBM = 'WEBM',
  OGG = 'OGG',
}

export interface StreamInfo {
  streamUrl: string;
  track: Track;
  expiresAt?: number;
  format: AudioFormat;
}

File: backend/src/main/kotlin/com/musicstreaming/models/StreamInfo.kt
package com.musicstreaming.models

import kotlinx.serialization.Serializable

@Serializable
data class StreamInfo(
    val streamUrl: String,
    val track: Track,
    val expiresAt: Long? = null,
    val format: AudioFormat
) {
    init {
        require(streamUrl.isNotBlank()) { "Stream URL cannot be blank" }
    }
}

@Serializable
enum class AudioFormat {
    MP3,
    AAC,
    WEBM,
    OGG
}
